cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/client/AuthRequestFilter.java

```diff
@@ -0,0 +1,77 @@
+package org.bf2.cos.fleetshard.sync.client;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import org.bf2.cos.fleetshard.support.metrics.MetricsRecorder;
+import org.bf2.cos.fleetshard.sync.FleetShardSyncConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.quarkus.oidc.client.filter.OidcClientRequestFilter;
+import io.quarkus.oidc.client.runtime.DisabledOidcClientException;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+
+@ApplicationScoped
+public class AuthRequestFilter implements ClientRequestFilter, ClientResponseFilter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OidcClientRequestFilter.class);
+    private static final String BEARER_SCHEME_WITH_SPACE = OidcConstants.BEARER_SCHEME + " ";
+    private static final String METRICS_INVOKE = "connectors.oidc.invoke";
+
+    @Inject
+    FleetShardSyncConfig config;
+    @Inject
+    MeterRegistry registry;
+    @Inject
+    AuthTokenSupplier tokenSupplier;
+
+    private MetricsRecorder recorderInvoke;
+
+    @PostConstruct
+    public void init() {
+        this.recorderInvoke = MetricsRecorder.of(registry, config.metrics().baseName() + "." + METRICS_INVOKE);
+    }
+
+    @Override
+    public void filter(ClientRequestContext requestContext) throws IOException {
+        this.recorderInvoke.record(
+            () -> requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, BEARER_SCHEME_WITH_SPACE + tokenSupplier.get()),
+            e -> {
+                if (e instanceof DisabledOidcClientException) {
+                    LOGGER.warn(
+                        "OidcClient is not is not enabled, aborting the request with HTTP 500 error: {}}",
+                        e.getMessage());
+
+                    requestContext.abortWith(Response.status(500).build());
+                } else {
+                    LOGGER.warn(
+                        "Access token is not available, aborting the request with HTTP 401 error: {}}",
+                        e.getMessage());
+
+                    requestContext.abortWith(Response.status(401).build());
+                }
+
+                tokenSupplier.reset();
+            });
+    }
+
+    @Override
+    public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext) throws IOException {
+        final int code = responseContext.getStatus();
+        final boolean success = code / 100 == 2;
+
+        if (!success && code != 410) {
```

Comment:
We seem to be recreating the client on everything different from 2xx or 410. Is this intended?

Reply:
Yes as for some auth issues, the error is just 404 or a 500x but we don't know exactly if it is an issue with the token or what not. 

Do you think there is anything better ?

Reply:
@rinaldodev fixed